evcache-core/src/main/java/com/netflix/evcache/EVCacheImpl.java

```diff
@@ -2210,12 +2212,27 @@ public int getDefaultTTL() {
         return _timeToLive;
     }
 
+    private EVCacheClient[] getClientsForWrite() {
+        EVCacheClient[] clients;
+        if (clientWriteToAllReplicas.get()) {
```

Comment:
If router instance is down or the tcp connection is severed, we need to handle the retries. For now, lets push these changes and release a candidate build.